acme/src/main/java/io/micronaut/acme/services/AcmeService.java

```diff
@@ -138,6 +138,28 @@ public X509Certificate getCurrentCertificate() {
         }
     }
 
+    /**
+     * Returns the full certificate chain.
+     *
+     * @return array of each of the certificates in the chain
+     */
+    protected X509Certificate[] getFullCertificateChain() {
```

Comment:
this should never be null

Reply:
Actually it can in the exceptional case right now. Thinking that should change and make it so that it never can be null.

Reply:
Made it so that even the exception case returns an empty array. [e960455](https://github.com/micronaut-projects/micronaut-acme/pull/161/commits/e96045515e7a2ff8239505cbe82abd2435e8983f)

Reply:
Thanks, annotate with `@NonNull` now that it is never null.

Reply:
Added here [67394be](https://github.com/micronaut-projects/micronaut-acme/pull/161/commits/67394be7d076e149a7c4b43006f02457ce9850c7)